demo/src/app/shared/analytics/analytics.ts

```diff
@@ -0,0 +1,39 @@
+import {Injectable} from '@angular/core';
+import {Router, NavigationEnd} from '@angular/router';
+import {Location} from '@angular/common';
+
+declare const ga: any;
+
+/**
+ * Simple Google Analytics service. Note that all its methods don't do anything unless the app
+ * is deployed on ng-bootstrap.github.io. This avoids sending events and page views during development.
+ */
+@Injectable()
+export class Analytics {
+  private _enabled: boolean;
+
+  constructor(private _location: Location, private _router: Router) {
+    this._enabled = window.location.href.indexOf('ng-bootstrap.github.io') >= 0;
+  }
+
+  /**
+   * Intended to be called only once. Subscribes to router events and sends a page view
+   * after each ended navigation event.
+   */
+  trackPageViews() {
```

Comment:
First, I just thought injecting the service was necessary to instantiate it, and injecting it to do nothing with it looked weird. So I thought leaving the responsibility to the main component to decide to start tracking page views looked better to me.

Second, although this is untested, not putting this logic in the constructor makes it more easily testable. Basically the same reason why the angular doc encourages the use of ngOnInit rather than the constructor to start interacting with services injected in the constructor: https://angular.io/docs/ts/latest/tutorial/toh-pt4.html#!#the-ngoninit-lifecycle-hook